plugins/jobs/latestBuild.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const boom = require('boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const idSchema = joi.reach(schema.models.job.base, 'id');
+const statusSchema = joi.reach(schema.models.build.base, 'status');
+
+module.exports = () => ({
+    method: 'GET',
+    path: '/jobs/{id}/latestBuild',
+    config: {
+        description: 'Get latest build for a given job',
+        notes: 'Return latest build of status specified',
+        tags: ['api', 'job', 'build'],
+        auth: {
+            strategies: ['token'],
+            scope: ['user', 'build', 'pipeline']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const jobFactory = request.server.app.jobFactory;
+            const { status } = request.query || {};
+
+            return jobFactory.get(request.params.id)
+                .then((job) => {
+                    if (!job) {
+                        throw boom.notFound('Job does not exist');
+                    }
+
+                    return job.getLatestBuild({ status });
+                })
+                .then((build) => {
+                    if (Object.getOwnPropertyNames(build).length === 0) {
+                        return reply({});
```

Comment:
I'm also unsure whether to return `{}` or `notFound` here. Want to hear others' opinion on this

Reply:
Not sure as well, but I'm okay with `{}`

Reply:
I think 404 is semantically correct. With empty object, a client now will have to check not only http status but also look into the return data to see if there is actual result  404 eliminates this problem.

Reply:
I would say return 404 with body saying `{status: "build does not exist"}` (consistent with responses in other places)